frontend/src/ui/Access.tsx

```diff
@@ -0,0 +1,698 @@
+            });
+        };
+
+        const handleCreate = (inputValue: string) => {
+            if (!inputValue.startsWith("ROLE_")) {
+                return;
+            }
+            const newRole: Option = {
+                value: inputValue,
+                label: formatUnknownRole(inputValue),
+            };
+            setSelection(prev => [...prev, newRole]);
+
+            setRoleSelections({
+                ...roleSelections,
+                readRoles: [...roleSelections.readRoles, inputValue],
+            });
+        };
+
+        const handleChange = (choice: MultiValue<Option>) => {
+            const newRoles = choice
+                .filter(option => !selection.includes(option))
+                .map(option => option.value);
+
+            setSelection([...choice].sort(roleComparator));
+            setOptions(prev => prev.filter(
+                option => !choice.some(opt => opt.value === option.value)
+            ));
+            setRoleSelections({
+                ...roleSelections,
+                readRoles: [...roleSelections.readRoles, ...newRoles],
+            });
+        };
+
+        const handlePaste = (event: React.ClipboardEvent<HTMLInputElement>) => {
+            event.preventDefault();
+
+            if (kind === "User") {
+                const clipboardData = event.clipboardData.getData("Text");
+                const names = clipboardData.split("\n").map(name => name.trim());
+
+                const optionsToAdd: Option[] = names
+                    .map(name => options.filter(option => option.label === name)[0])
+                    .filter(option => option !== undefined);
+
+                if (optionsToAdd.length > 0) {
+                    handleChange([...selection, ...optionsToAdd]);
+                    setMenuIsOpen(false);
+                }
+            }
+        };
+
+
+        return <div css={{
+            flex: "1 1 320px",
+            display: "flex",
+            flexDirection: "column",
+            maxWidth: 900,
+        }}>
+            <h4>{translations.heading}</h4>
+            <div onPaste={handlePaste}>
+                <CreatableSelect
+                    onMenuOpen={() => setMenuIsOpen(true)}
+                    onMenuClose={() => setMenuIsOpen(false)}
+                    controlShouldRenderValue={false}
+                    isClearable={false}
+                    isMulti
+                    isSearchable
+                    placeholder={translations.placeholder}
+                    formatCreateLabel={input =>
+                        /^ROLE_\w+/.test(input) && t("manage.access.select.create", { item: input })
+                    }
+                    value={selection}
+                    onCreateOption={handleCreate}
+                    filterOption={(option, inputValue) => !!option.label
+                        && option.label.toLowerCase().includes(inputValue.toLowerCase())
+                    }
```

Comment:
I don't ... think so? I specifically want to be able to get partial matches with this and I don't believe `localeCompare` allows for that. I know this is the default behaviour of react-select but the filter is still needed so it won't show an empty box like this:
<img width="539" alt="Bildschirmfoto 2023-09-19 um 10 13 05" src="https://github.com/elan-ev/tobira/assets/94838646/9ab55b0b-1fca-4638-8f35-ad711c4f5b48">
Usually an unknown input would show the `create` option, but since this frankenstein's monster uses a custom `onCreateOption` function that only allows certain strings, I think it needs to be done like I coded it.

Reply:
Oh dang, yes, my mistake. I looked around for how to do this better than just calling `toLowerCase`: see [this stackoverflow](https://stackoverflow.com/questions/39548303/javascript-string-prototype-contains-with-locale). The top answer seems decent.

Reply:
Oh ok, I've seen that post too but didn't think it was worth writing an extra function. But if you say that's the way to go, I'll do it of course.